qlty-config/src/config/builder.rs

```diff
@@ -355,9 +358,97 @@ impl Builder {
     }
 }
 
+fn merge_enabled_plugins(existing: &EnabledPlugin, new: &EnabledPlugin) -> EnabledPlugin {
+    debug!(
+        "Merging enabled plugins: existing: {:?}, new: {:?}",
+        existing, new
+    );
+
+    if existing.version != new.version {
+        warn!(
+            "Merging enabled plugins with different versions: {} and {}, using {}.",
+            existing.version, new.version, new.version
+        );
+    }
+
+    let merged_mode = existing.mode.unwrap_or(new.mode.unwrap_or_default());
+
+    EnabledPlugin {
+        name: existing.name.clone(),
+        prefix: existing.prefix.clone(),
+        mode: Some(merged_mode),
+        version: new.version.clone(),
+        triggers: existing
+            .triggers
+            .iter()
+            .chain(new.triggers.iter())
+            .cloned()
+            .collect(),
+        skip_upstream: existing.skip_upstream.or(new.skip_upstream),
+        package_file: existing.package_file.clone().or(new.package_file.clone()),
+        fetch: existing
+            .fetch
```

Comment:
So the merging behaviour is a bit tricky for strings and enums that can hold only one value, problem being which one is to be selected, since this happens after toml merging, right now no way to know which came from where, also someone can have two different enabled plugins blocks for same plugin name and prefix defined in their `qlty.toml` file similar to the integration test right now.

For array attributes I just merged the arrays.